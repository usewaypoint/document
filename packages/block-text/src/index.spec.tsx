import React from 'react';

import { render } from '@testing-library/react';

import { Text } from '.';

describe('block-text', () => {
  it('renders with default values', () => {
    expect(render(<Text />).asFragment()).toMatchSnapshot();
  });
});
