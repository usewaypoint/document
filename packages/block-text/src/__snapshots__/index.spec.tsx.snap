// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block-text renders with default values 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`block-text renders with safe markdown 1`] = `
<DocumentFragment>
  <div>
    <p>
      This 
      <span>
        text
      </span>
       block has the 
      <strong>
        Markdown
      </strong>
       option 
      <em>
        turned on
      </em>
      .
    </p>
    

    <ul>
      

      <li>
        One
      </li>
      

      <li>
        Two
      </li>
      

      <li>
        Three
      </li>
      

    </ul>
    

    <p>
      Powered by 
      <a
        href="https://usewaypoint.com"
        target="_blank"
      >
        Waypoint
      </a>
    </p>
    

  </div>
</DocumentFragment>
`;

exports[`block-text renders without markdown 1`] = `
<DocumentFragment>
  <div>
    ## This is not &lt;span&gt;markdown&lt;/span&gt;
  </div>
</DocumentFragment>
`;

exports[`block-text sanitizes HTML 1`] = `
<DocumentFragment>
  <div>
    

    <img
      src="x"
    />
    


    <p>
      <a
        target="_blank"
      >
        a
      </a>
      <br />
      <a
        target="_blank"
      >
        Basic
      </a>
      <br />
      <a
        target="_blank"
      >
        Local Storage
      </a>
      <br />
      <a
        target="_blank"
      >
        CaseInsensitive
      </a>
      <br />
      <a
        target="_blank"
      >
        URL
      </a>
    </p>
    

    <p>
      <a
        target="_blank"
      >
        In Quotes
      </a>
      <br />
      [a](j a v a s c r i p t:prompt(document.cookie))
      <br />
      <a
        target="_blank"
      >
        a
      </a>
      <br />
      <a
        target="_blank"
      >
        a
      </a>
      <br />
      <img
        alt="Uh oh..."
        src="%22onerror=%22alert('XSS')"
      />
      <br />
      <img
        alt="Uh oh..."
        src="https://www.example.com/image.png%22onload=%22alert('XSS')"
      />
      <br />
      <img
        alt="Escape SRC - onload"
        src="https://www.example.com/image.png%22onload=%22alert('ImageOnLoad')"
      />
      <br />
      <img
        alt="Escape SRC - onerror"
        src="%22onerror=%22alert('ImageOnError')"
      />
    </p>
    

  </div>
</DocumentFragment>
`;
